src/mlpack/methods/lsh/lsh_search_impl.hpp

```diff
@@ -362,27 +652,65 @@ void LSHSearch<SortPolicy>::ReturnIndicesFromTable(
 
   // Compute the projection of the query in each table.
   arma::mat allProjInTables(numProj, numTablesToSearch);
+  arma::mat queryCodesNotFloored(numProj, numTablesToSearch);
   for (size_t i = 0; i < numTablesToSearch; i++)
-    //allProjInTables.unsafe_col(i) = projections[i].t() * queryPoint;
-    allProjInTables.unsafe_col(i) = projections.slice(i).t() * queryPoint;
-  allProjInTables += offsets.cols(0, numTablesToSearch - 1);
-  allProjInTables /= hashWidth;
+    queryCodesNotFloored.unsafe_col(i) = projections.slice(i).t() * queryPoint;
+  queryCodesNotFloored += offsets.cols(0, numTablesToSearch - 1);
+  allProjInTables = arma::floor(queryCodesNotFloored/hashWidth);
 
   // Compute the hash value of each key of the query into a bucket of the
   // 'secondHashTable' using the 'secondHashWeights'.
-  arma::rowvec hashVec = secondHashWeights.t() * arma::floor(allProjInTables);
+  arma::rowvec hashVec = secondHashWeights.t() * allProjInTables;
 
+  // mod and floor hashVec to compute 2nd-level codes
   for (size_t i = 0; i < hashVec.n_elem; i++)
     hashVec[i] = (double) ((size_t) hashVec[i] % secondHashSize);
 
   Log::Assert(hashVec.n_elem == numTablesToSearch);
 
+  // Compute hashVectors of additional probing bins
+  arma::mat hashMat;
```

Comment:
Two thoughts:

 * couldn't this be an `arma::Mat<size_t>`?  I'm not sure why the original `hashVec` didn't hold a `size_t` internally.
 * we could probably save a copy and speed things up a little if we never use the original `hashVec` and instead use `hashMat` and set its size to `(T + 1, numTablesToSearch)`.